services/QuillLMS/app/lib/synthetic/generators/base.rb

```diff
@@ -0,0 +1,41 @@
+# frozen_string_literal: true
+
+module Synthetic
+  module Generators
+    class Base < ApplicationService
+
+      class NotImplementedError < StandardError; end
+
+      attr_reader :results_hash, :languages
+
+      # takes an array of unique strings, e.g. ['original string', 'sample 2']
+      def initialize(string_array, options = {})
+        # create and hash with each string  to an empty hash, e.g.
+        # {'string' => {}, 'string 2' => {}
+        @results_hash = string_array.map {|string| [string,{}] }.to_h
+        @languages = options[:languages] || Synthetic::Generators::Translation::TRAIN_LANGUAGES
```

Comment:
Not a huge deal, but should `languages` be part of the Base model given that it's not relevant to all children?

Reply:
That's fair. I was trying to avoid having to overwrite `initialize` in the subclasses by setting all options (used or unused) in the `base` class. But, you're right, it does feel a little out of place. I've moved the initialization to the subclass (and updated the instructions).